midterm-2023-11-07/code/5037-stefani-georgievska/script.ts

```diff
@@ -0,0 +1,137 @@
+interface Movies {
+    id: number;
+    title: string;
+    director: string;
+    year: number;
+    genre: string;
+    plot: string;
+    cast: string[];
+    oscars: number;
+}
+type MoviesSorter = (first: Movies, second: Movies) => number;
+
+document.addEventListener("DOMContentLoaded", siteCode);
+
+let movies: Movies[] = [];
+
+async function siteCode() {
+        const data = await loadData();
+        movies = data;
+
+    displayMovies(movies);
+
+    const titleSort = document.getElementById("sort-title")!;
+    titleSort.addEventListener("click", sortByTitle);
+
+    const idSort = document.getElementById("sort-id")!;
+    idSort.addEventListener("click", sortById);
+
+    const applyFilterButton = document.getElementById("apply-filter")!;
+    applyFilterButton.addEventListener("click", applyFilter)
+
+    const modal = document.getElementById("biblio-details")!;
+    modal.addEventListener("click", () => {
+        modal.classList.add("hidden");
+    });
+
+}
+
+const titleSorter: MoviesSorter = (first, second) => first.title.localeCompare(second.title);
+const idSorter: MoviesSorter = (first, second) => first.id - second.id;
+
+const sortByTitle = () => {
+    const sortedMovies = movies.toSorted(titleSorter);
+    displayMovies(sortedMovies);
+}
+
+const sortById = () => {
+    const sortedMovies = movies.toSorted(idSorter);
+    displayMovies(sortedMovies);
+}
+
+const applyFilter = () => {
+    const titleElement = document.getElementById("title-filter") as HTMLSelectElement;
```

Comment:
Neither the title nor the director as select elements, they're simple text inputs :)
+5 for getting the values, though